src/Plugin/Replace/Handler.php

```diff
@@ -57,106 +80,103 @@ public function run(): Task {
 	 */
 	private static function getRecordValues(Client $manticoreClient, Payload $payload, array $fields): array {
-		$sql = 'SELECT * FROM '.$payload->table.' WHERE id = '.$payload->id;
+		$sql = 'SELECT * FROM ' . $payload->table . ' WHERE id = ' . $payload->id;
 
+		/** @var array<int, array<string, array<int, array<string, string>>>> $records */
 		$records = $manticoreClient
 			->sendRequest($sql)
 			->getResult();
 
-		$mvaFields = [];
-
-		foreach ($fields as $key) {
-			if ($key['Type'] !== 'mva') {
-				continue;
-			}
-
-			$mvaFields[] = $key['Field'];
+		if (isset($records[0]['data'][0])) {
+			return self::morphValuesByFieldType($records[0]['data'][0], $fields);
 		}
 
-		if (is_array($records) && !empty($records[0]['data'][0])) {
-			// We need migrate MVA values to correct syntax for replace call
-			if ($mvaFields !== []) {
-				foreach ($mvaFields as $field) {
-					$records[0]['data'][0][$field] = '('.$records[0]['data'][0][$field].')';
-				}
-			}
+		return ['id' => $payload->id];
+	}
 
-			return $records[0]['data'][0];
+	/**
+	 * @param array<string, bool|float|int|string|array<int, string>> $records
+	 * @param array<int|string, array{type:string, properties:string}> $fields
+	 * @return array<string, bool|float|int|string>
+	 */
+	private static function morphValuesByFieldType(array $records, array $fields): array {
+
+		foreach ($records as $fieldName => $fieldValue) {
+			$records[$fieldName] = match ($fields[$fieldName]['type']) {
+				self::TYPE_INT, self::TYPE_BIGINT, self::TYPE_TIMESTAMP => (int)$fieldValue,
+				self::TYPE_BOOL => (bool)$fieldValue,
+				self::TYPE_FLOAT => (float)$fieldValue,
+				self::TYPE_TEXT, self::TYPE_STRING, self::TYPE_JSON =>
+					"'" . (is_array($fieldValue) ? json_encode($fieldValue) : $fieldValue)  . "'",
+				self::TYPE_MVA, self::TYPE_MVA64, self::TYPE_FLOAT_VECTOR =>
+					'(' . (is_array($fieldValue) ? implode(',', $fieldValue) : $fieldValue) . ')',
+				default => $fieldValue
+			};
 		}
 
-		return ['id' => $payload->id];
+		/** @var array<string, bool|float|int|string> */
+		return $records;
 	}
 
-
 	/**
-	 * @param  string  $tableName
-	 * @param  array<string, string|int|array<string>> $set
+	 * @param string $tableName
+	 * @param array<string, bool|float|int|string> $set
 	 * @return string
 	 */
 	private static function buildQuery(string $tableName, array $set): string {
-		$keys = [];
-		$values = [];
-		foreach ($set as $key => $value) {
-			$keys[] = $key;
-
-			if (is_numeric($value)) {
-				$values[] = $value;
-			} elseif (isset($value[0]) && is_string($value) && $value[0] === '(') {
-				$values[] = $value;
-			} elseif (is_array($value)) {
-				$values[] = '('.implode(',', $value).')';
-			} else {
-				$values[] = "'".$value."'";
-			}
-		}
-
-		return 'REPLACE INTO `'.$tableName.'` ('.implode(',', $keys).') '.
-			'VALUES ('.implode(',', $values).')';
+		return 'REPLACE INTO `' . $tableName . '` (' . implode(',', array_keys($set)) . ') ' .
```

Comment:
It's good idea to move implode out of concatenation and use double quotes with interpolation.